### PR TITLE
ttf Font implementation

### PR DESCRIPTION
**Pull request by mwitchwilliams**
GearVRf-DCO-1.0-Signed-off-by: Mitch Williams
m1.williams@partner.samsung.com

Added DEF/USE capability to X3D &lt;FontStyle&gt; node, so Font Styles can be shared by text within an X3D file.
Addressed issues presented in 'ww_font_PullReq' pull request by implementing a new Constructor in GVRTextViewSceneObject and an inner class 'Text_FontParams' in X3DObject.java.
Also implemented the gvrScene.setBackground() command per request from previous pull request.

This pull request supersedes 'ww_font_PullReq' and request that one be closed.

### CODE REVIEW

**Comment by mwitchwilliams** [comment]
X3D test cases with documentation and ,ttf font files can be found in the GVRf-Tests repository under x3d -> fonts

**Comment by liaxim** [comment]
This looks much better. Thanks.

Looking at the new ctor, the Builder pattern comes to mind: https://en.wikipedia.org/wiki/Builder_pattern.

File: GVRf/Framework/framework/src/main/java/org/gearvrf/scene_objects/GVRTextViewSceneObject.java [hunk 1]
```diff
@@ -224,6 +365,21 @@ public void setText(CharSequence text) {
         mIsChanged = true;
     }
 
+    public void setText(String text) {
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/scene_objects/GVRTextViewSceneObject.java]: This method is unnecessary. String implements CharSequence. We can remove later.

**Comment by liaxim** [comment]
I think the justification should have been kept inside x3d and translated by x3d to/from the Android Gravity flags instead of pushing this foreign concept into the TextViewSceneObject. We can live with it.

Planning to merge Thursday morning.

**Comment by liaxim** [comment]
Tests match the expected output but the test pull request need rebase.

**Pull request merged**